classes/Cart.php

```diff
@@ -799,6 +799,22 @@ public static function cacheSomeAttributesLists($ipa_list, $id_lang)
         }
     }
 
+    public function validateCartAddresses()
+    {
+        $needUpdate = false;
+        foreach (['invoice', 'delivery'] as $type) {
+            if (!Address::isValid($this->{'id_address_'.$type})) {
```

Comment:
Also it sound weird to save data in validateXX function, how about rename function `validateAndPersistAddresses` ?